test/rules/object-literal-shorthand/never/test.ts.lint

```diff
@@ -0,0 +1,62 @@
+const asyncFn = {
+  async f() {
+  ~~~~~~~~~~~
+    await some_promise;
+~~~~~~~~~~~~~~~~~~~~~~~
+  },
+~~~ [OBJECT_LITERAL_DISALLOWED]
+  async* fa() {
+  ~~~~~~~~~~~~~
+    await some_promise;
+~~~~~~~~~~~~~~~~~~~~~~~
+  }
+~~~ [OBJECT_LITERAL_DISALLOWED]
+};
+
+const bad = {
+  w() {},
+  ~~~~~~ [OBJECT_LITERAL_DISALLOWED]
+  *x() {},
+  ~~~~~~~ [OBJECT_LITERAL_DISALLOWED]
+  [y]() {},
+  ~~~~~~~~ [OBJECT_LITERAL_DISALLOWED]
+  z,
+  ~ [OBJECT_LITERAL_DISALLOWED]
+  nest: {
+    nestBad() {},
+    ~~~~~~~~~~~~ [OBJECT_LITERAL_DISALLOWED]
+    nextGood: function(prop: string): void {}
+  }
+};
+
+const good = {
+  w: function() {},
+  x: function *() {},
+  [y]: function() {}
+};
+
+const arrows = {
+  x: (y) => y  // this is OK.
+};
+
+const namedFunctions = {
+  x: function y() {}  // named function expressions are also OK.
+};
+
+const quotes = {
+  "foo-bar": function() {},
+  "foo-bar"() {}
+  ~~~~~~~~~~~~~~ [OBJECT_LITERAL_DISALLOWED]
+};
+
+const extraCases = {
+  x,
+  ~ [OBJECT_LITERAL_DISALLOWED]
+  a: 123,
+  b: "hello",
+  c: 'c',
+  ["a" + "nested"]: {
+    x: x
+  }
+};
+[OBJECT_LITERAL_DISALLOWED]: Shorthand property assignments have been disallowed.
```

Comment:
please add a test case with class methods. these should be ignored by the rule

Reply:
You mean class members which instantiate object literals with shorthand props?

Reply:
I mean regular method declarations inside a class. They are ignored by the rule. I just want a test to ensure this won't regress.

```ts
class C {
    doStuff() {} // shouldn't be an error
}
```

Reply:
Ahh of course. Thanks